pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -199,7 +217,12 @@ func (this *SnapshotManager) CreateSnapshot(peID astrolabe.ProtectedEntityID, ta
 	return updatedPeID, nil
 }
 
-func (this *SnapshotManager) DeleteSnapshot(peID astrolabe.ProtectedEntityID) error {
+func (this *SnapshotManager) DeleteSnapshotWithBackupRepository(peID astrolabe.ProtectedEntityID) error {
+	this.WithField("peID", peID.String()).Info("SnapshotManager.DeleteSnapshotWithBackupRepository was called.")
+	return this.DeleteSnapshot(peID, true)
+}
+
+func (this *SnapshotManager) DeleteSnapshot(peID astrolabe.ProtectedEntityID, useBackupRepository bool) error {
```

Comment:
Should this be deleteSnapshot as it should be an internal function?

Reply:
volume_snapshotter_plugin.go also calls this function, so keep DeleteSnapshot here.

Reply:
Change to deleteSnapshot as a internal help function.